src/ioctls.cc

```diff
@@ -1,39 +1,88 @@
-#include <nan.h>
+#include <node_api.h>
+#include <assert.h>
 #include <sys/ioctl.h>
 #include <linux/input.h>
 #include <fcntl.h>
 #include <errno.h>
 #include <stdio.h>
 #include <string.h>
 
-using v8::FunctionTemplate;
-using v8::Handle;
-using v8::Object;
-using v8::String;
-using Nan::GetFunction;
-using Nan::New;
-using Nan::Set;
-using Nan::To;
+napi_value EvdevNewFromFd(napi_env env, napi_callback_info info) {
+  napi_status status;
 
+  size_t argc = 1;
+  napi_value args[1];
+  status = napi_get_cb_info(env, info, &argc, args, nullptr, nullptr);
+  assert(status == napi_ok);
+
+  if (argc < 1) {
+    napi_throw_type_error(env, nullptr, "Wrong number of arguments");
+    return nullptr;
+  }
+
+  napi_valuetype valuetypeArg0;
+  status = napi_typeof(env, args[0], &valuetypeArg0);
+  assert(status == napi_ok);
+
+  if (valuetypeArg0 != napi_number) {
+    napi_throw_type_error(env, nullptr, "Wrong arguments");
+    return nullptr;
+  }
+
+  int fd;
+  napi_get_value_int32(env, args[0], &fd);
+  assert(status == napi_ok);
 
-NAN_METHOD(evdev_new_from_fd) {
   int rc;
   struct input_id id;
-  rc = ioctl(To<int>(info[0]).FromJust(), EVIOCGID, &id);
+  rc = ioctl(fd, EVIOCGID, &id);
   if(rc == -1 ){
-    Nan::ThrowError(strerror(errno));
-    //Nan::ThrowError(info[0]);
+    napi_throw_type_error(env, nullptr, strerror(errno));
+    return nullptr;
   }
-  v8::Local<v8::Object> size = Nan::New<v8::Object>();
-  Nan::Set(size,Nan::New("bustype").ToLocalChecked(), Nan::New<v8::Number>((double)id.bustype));
-  Nan::Set(size,Nan::New("vendor").ToLocalChecked(), Nan::New<v8::Number>((double)id.vendor));
-  Nan::Set(size,Nan::New("product").ToLocalChecked(), Nan::New<v8::Number>((double)id.product));
-  Nan::Set(size,Nan::New("version").ToLocalChecked(), Nan::New<v8::Number>((double)id.version));
-  info.GetReturnValue().Set(size);
+
+  napi_value size;
+  status = napi_create_object(env, &size);
+  assert(status == napi_ok);
+
+  napi_value asdf;
+  status = napi_create_double(env, (double)id.bustype, &asdf);
+  assert(status == napi_ok);
+  status = napi_set_named_property(env, size, "bustype", asdf);
+  assert(status == napi_ok);
+
+  status = napi_create_double(env, (double)id.vendor, &asdf);
+  assert(status == napi_ok);
+  status = napi_set_named_property(env, size, "vendor", asdf);
+  assert(status == napi_ok);
+
+  status = napi_create_double(env, (double)id.product, &asdf);
+  assert(status == napi_ok);
+  status = napi_set_named_property(env, size, "product", asdf);
+  assert(status == napi_ok);
+
+  status = napi_create_double(env, (double)id.version, &asdf);
+  assert(status == napi_ok);
+  status = napi_set_named_property(env, size, "version", asdf);
+  assert(status == napi_ok);
```

Comment:
oof, I forgot to clean this up after it was working :( 
```suggestion
  napi_value valueTransfer;
  status = napi_create_double(env, (double)id.bustype, &valueTransfer);
  assert(status == napi_ok);
  status = napi_set_named_property(env, size, "bustype", valueTransfer);
  assert(status == napi_ok);

  status = napi_create_double(env, (double)id.vendor, &valueTransfer);
  assert(status == napi_ok);
  status = napi_set_named_property(env, size, "vendor", valueTransfer);
  assert(status == napi_ok);

  status = napi_create_double(env, (double)id.product, &valueTransfer);
  assert(status == napi_ok);
  status = napi_set_named_property(env, size, "product", valueTransfer);
  assert(status == napi_ok);

  status = napi_create_double(env, (double)id.version, &valueTransfer);
  assert(status == napi_ok);
  status = napi_set_named_property(env, size, "version", valueTransfer);
  assert(status == napi_ok);
```
sorry for the churn